xbmc/windowing/osx/WinSystemOSX.mm

```diff
@@ -409,20 +418,19 @@ void DisplayFadeFromBlack(CGDisplayFadeReservationToken fade_token, bool fade)
 
 NSString* screenNameForDisplay(CGDirectDisplayID displayID)
 {
-  NSAutoreleasePool* pool = [[NSAutoreleasePool alloc] init];
-
-  NSString *screenName = nil;
-
-  NSDictionary *deviceInfo = (NSDictionary *)IODisplayCreateInfoDictionary(CGDisplayIOServicePort(displayID), kIODisplayOnlyPreferredName);
-  NSDictionary *localizedNames = [deviceInfo objectForKey:[NSString stringWithUTF8String:kDisplayProductName]];
-
-  if ([localizedNames count] > 0)
+  NSString* screenName;
+  @autoreleasepool
   {
-    screenName = [[localizedNames objectForKey:[[localizedNames allKeys] objectAtIndex:0]] retain];
-  }
+    NSDictionary* deviceInfo = (__bridge_transfer NSDictionary*)IODisplayCreateInfoDictionary(
```

Comment:
__bridge_transfer is not a cast, this here is objc :)

Reply:
just in case verified with the following code:

```objc++
#import <Foundation/Foundation.h>
#include <CoreFoundation/CoreFoundation.h>
#include <iostream>

int main(int argc, char const *argv[])
{
  std::cout << @"asd".UTF8String << "\n";
  NSLog(@"C-cast %@", (__bridge NSString*)kCFBundleVersionKey);
  NSLog(@"C++-cast %@", static_cast<__bridge NSString*>(kCFBundleVersionKey));
  return 0;
}
```

```
❯ clang -fobjc-arc -lc++ untitled.mm
untitled.mm:9:37: error: type name requires a specifier or qualifier
  NSLog(@"C++-cast %@", static_cast<__bridge NSString*>(kCFBundleVersionKey));
                                    ^
untitled.mm:9:37: error: expected '>'
  NSLog(@"C++-cast %@", static_cast<__bridge NSString*>(kCFBundleVersionKey));
                                    ^
                                    >
untitled.mm:9:36: note: to match this '<'
  NSLog(@"C++-cast %@", static_cast<__bridge NSString*>(kCFBundleVersionKey));
                                   ^
2 errors generated.
```

Reply:
https://developer.apple.com/library/archive/documentation/CoreFoundation/Conceptual/CFDesignConcepts/Articles/tollFreeBridgedTypes.html

welcome to ARC and objc :)